t/integration/test_batches.py

```diff
@@ -150,6 +150,7 @@ def test_signals(celery_app: Celery, celery_worker: TestWorkController) -> None:
         # Other task signals are not implemented.
         (signals.task_retry, 0),
         (signals.task_success, 1),
+        (signals.task_received, 3),
```

Comment:
I think we'll also want to assert the other signals.

It *might* make sense to keep the PRs small and implement them one at a time?

Reply:
I can split the PR into more small ones (one for received, and one for success).

Reply:
I'd say that if you know `task_received` works already, let's get that one done and then move onto the others?